.gitlab-ci.yml

```diff
@@ -79,23 +81,21 @@ check:nix-dry:
 
 check:test-generate:
   stage: check
-  interruptible: true
+  needs: []
   script:
     - >
       nix-shell --run '
-      ./scripts/test-pipelines.sh > ./tmp/test-pipelines.yml
+      ./scripts/check-test-generate.sh > ./tmp/check-test.yml
       '
   artifacts:
     when: always
     paths:
-      - ./tmp/test-pipelines.yml
+      - ./tmp/check-test.yml
   rules:
-    # Runs on feature and staging commits and ignores version commits
-    - if: $CI_COMMIT_BRANCH =~ /^(?:feature.*|staging)$/ && $CI_COMMIT_TITLE !~ /^[0-9]+\.[0-9]+\.[0-9]+(?:-.*[0-9]+)?$/
-    # Runs on tag pipeline where the tag is a prerelease or release version
-    - if: $CI_COMMIT_TAG =~ /^v[0-9]+\.[0-9]+\.[0-9]+(?:-.*[0-9]+)?$/
-    # Manually run on commits other than master and ignore version commits
-    - if: $CI_COMMIT_BRANCH && $CI_COMMIT_BRANCH != 'master' && $CI_COMMIT_TITLE !~ /^[0-9]+\.[0-9]+\.[0-9]+(?:-.*[0-9]+)?$/
+    # Runs on staging commits and ignores version commits
+    - if: $CI_COMMIT_BRANCH =~ /^feature.*$/ && $CI_COMMIT_TITLE !~ /^[0-9]+\.[0-9]+\.[0-9]+(?:-.*[0-9]+)?$/
+    # Manually run on commits other than master and staging and ignore version commits
+    - if: $CI_COMMIT_BRANCH && $CI_COMMIT_BRANCH !~ /^(?:master|staging)$/ && $CI_COMMIT_TITLE !~ /^[0-9]+\.[0-9]+\.[0-9]+(?:-.*[0-9]+)?$/
       when: manual
```

Comment:
Is this missing the original rules?

* Run on feature and staging commits and ignores version commits - why does it only say `feature.*` now?
* Runs on tag pipeline with the tag is a prerelease or release version - where did this go?
* The third rule is now `master|staging`, why did this change?

Reply:
When I originally brought these pipelines in from ts-demo-lib (back when we first created the staging branch here), I had pipelines running on staging/feature branches that weren't supposed to just so that I could check they were working, and staging got merged into master before I could fix it. Everything should now be the same as on ts-demo-lib (including the comments not always aligning with the actual rules, I just copy pasted from ts-demo-lib)

Reply:
Ok I see it now, TS demo lib has the correct rule set here. It should only be `feature.*` for the first rule, and then `master|staging` for second rule.